backend/tests/integration/tests/test_aws_iotcore.py

```diff
@@ -0,0 +1,443 @@
+            .call("POST", iot.URL_INTEGRATIONS, integration)
+        )
+        return response
+
+    def get_integrations(self, user: User) -> Response:
+        response = (
+            self.aws_api.with_auth(user.utoken)
+            .with_header("Content-Type", "application/json")
+            .call("GET", iot.URL_INTEGRATIONS)
+        )
+        return response
+
+    def check_integrations(self, user: User, expected_integration: Dict):
+        """Make sure iot-manager properly saves connection strings in its database."""
+        response = self.save_integration(user, expected_integration)
+        assert response.status_code == 201
+        self.logger.info("saved integrations")
+
+        self.logger.info("getting integrations")
+        response = self.get_integrations(user)
+        assert response.status_code == 200
+        self.logger.info(f"got integrations: {response.text}")
+        integrations = response.json()
+        assert len(integrations) > 0
+        assert "credentials" in integrations[0].keys()
+        assert "aws" in integrations[0]["credentials"].keys()
+        assert "access_key_id" in integrations[0]["credentials"]["aws"].keys()
+        assert (
+            integrations[0]["credentials"]["aws"]["access_key_id"] == AWS_ACCESS_KEY_ID
+        )
+        # the service will mask the AWS secret access key
+        assert "secret_access_key" in integrations[0]["credentials"]["aws"].keys()
+        assert (
+            integrations[0]["credentials"]["aws"]["secret_access_key"]
+            != AWS_SECRET_ACCESS_KEY
+        )
+        assert "region" in integrations[0]["credentials"]["aws"].keys()
+        assert integrations[0]["credentials"]["aws"]["region"] == AWS_REGION
+        assert "device_policy_name" in integrations[0]["credentials"]["aws"].keys()
+        assert (
+            integrations[0]["credentials"]["aws"]["device_policy_name"]
+            == AWS_DEVICE_POLICY_NAME
+        )
+
+
+class TestAWSIoTCoreIntegrations(_TestAWSIoTCoreBase):
+    @pytest.mark.parametrize(
+        "expected_integration",
+        [
+            {
+                "provider": "iot-core",
+                "credentials": {
+                    "type": "aws",
+                    "aws": {
+                        "access_key_id": AWS_ACCESS_KEY_ID,
+                        "secret_access_key": AWS_SECRET_ACCESS_KEY,
+                        "region": AWS_REGION,
+                        "device_policy_name": AWS_DEVICE_POLICY_NAME,
+                    },
+                },
+            },
+        ],
+    )
+    def test_get_and_set(self, clean_mongo, expected_integration):
+        """
+        Check that we can set and get integrations
+        """
+        self.logger.info("creating user in OS mode")
+        user = create_user_test_setup()
+        self.check_integrations(user, expected_integration)
+
+
+class TestAWSIoTCoreIntegrationsEnterprise(_TestAWSIoTCoreBase):
+    @pytest.mark.parametrize(
```

Comment:
This test also depends on the environment variables.
```suggestion
    @pytest.mark.skipif(
        not bool(os.environ.get("AWS_IOTCORE_ACCESS_KEY_ID")),
        reason="AWS_IOTCORE_ACCESS_KEY_ID not provided",
    )
    @pytest.mark.parametrize(
```